src/Gaffer/ScriptNode.cpp

```diff
@@ -534,48 +534,42 @@ void ScriptNode::paste( Node *parent )
 	}
 }
 
+namespace
+{
+
 void handleReconnectionOfOutputPlug( const ScriptNode *script, const DependencyNode *dependencyNode, const Plug *outputPlug )
 {
-	// for plugs that hold output plugs we need to recurse
-	if(outputPlug->typeId() == Plug::staticTypeId())
+	const Plug *inPlug = dependencyNode->correspondingInput( outputPlug );
+	if ( inPlug )
 	{
-		for( OutputPlugIterator it( outputPlug ); !it.done(); ++it )
+		const Plug *srcPlug = inPlug->getInput<Plug>();
+		if ( srcPlug )
 		{
-			handleReconnectionOfOutputPlug( script, dependencyNode, it->get() );
+			// record this plug's current outputs, and reconnect them. This is a copy of outputPlug->outputs() rather
+			// than a reference, as reconnection can modify outputPlug->outputs()...
+			Plug::OutputContainer outputs = outputPlug->outputs();
+			for ( Plug::OutputContainer::const_iterator oIt = outputs.begin(); oIt != outputs.end(); ++oIt )
+			{
+				Plug *dstPlug = *oIt;
+				if ( dstPlug && dstPlug->acceptsInput( srcPlug ) && script->isAncestorOf( dstPlug ) )
+				{
+					dstPlug->setInput( const_cast<Gaffer::Plug*>(srcPlug) );
+					return;
```

Comment:
If a parent plug has an input, then it is guaranteed that its children are connected to the equivalent children of that input. But here we're dealing with _outputs_, and whereas a plug can only have a single input, it can have many outputs. So, it's possible for the plug to have a single output at the parent level (with equivalent outputs at the child levels), but also have additional outputs from its children, that were made individually.